test/operations/wait.go

```diff
@@ -189,28 +193,56 @@ func grpcHealthyCondition(address string) (bool, error) {
 }
 
 // txMinedCondition
-func txMinedCondition(ctx context.Context, client *ethclient.Client, hash common.Hash) (bool, error) {
-	_, isPending, err := client.TransactionByHash(ctx, hash)
-	if err == ethereum.NotFound {
+func txMinedCondition(ctx context.Context, client ethClienter, hash common.Hash) (bool, error) {
+	// Get tx status
+	tx, isPending, err := client.TransactionByHash(ctx, hash)
+	if err == ethereum.NotFound || isPending {
 		return false, nil
 	}
-
 	if err != nil {
 		return false, err
 	}
+	// Check if tx has failed
+	receipt, err := client.TransactionReceipt(ctx, hash)
+	if err != nil {
+		return false, err
+	}
+	if receipt.Status == types.ReceiptStatusFailed {
+		// Get revert reason
+		reason, reasonErr := revertReason(ctx, client, tx, receipt.BlockNumber)
+		if reasonErr != nil {
+			reason = reasonErr.Error()
+		}
+		return false, fmt.Errorf("transaction has failed, reason: %s, receipt: %+v. tx: %+v", reason, receipt, tx)
+	}
+	return true, nil
+}
 
-	var done bool
-	if !isPending {
-		r, err := client.TransactionReceipt(ctx, hash)
+func revertReason(ctx context.Context, c ethClienter, tx *types.Transaction, blockNumber *big.Int) (string, error) {
```

Comment:
In state/errors.go there is a similar method to this: constructErrorFromRevert

Reply:
I think it's not similar enough to reuse it